packages/e2e-tests/specs/editor/various/theme-switching.test.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	activateTheme,
+	createNewPost,
+	deleteTheme,
+	insertBlock,
+	installTheme,
+	publishPost,
+	themeInstalled,
+} from '@wordpress/e2e-test-utils';
+
+const defaultTheme = 'twentytwenty';
+
+// These can be any themes available from wordpress.org that don't themselves cause editor errors.
+const testThemes = [ [ 'primer' ], [ 'astra' ] ];
+
+describe( 'Theme switching', () => {
```

Comment:
I know it was @gziolo's suggestion but i'm personally not entirely sold on the idea of this test as it relies on a external API that can break depending on the internet connection... (We don't have the same thing for plugins)

That said, we can give it a try and see after some time whether it's stable enough...

Reply:
Also, this would be better a test in Core e2e tests :)

Reply:
Yeah, I admit I had trouble figuring out where best to put this test, and where I decided to put it felt arbitrary. And then indeed this test is currently timing out in the build (while it passes locally) for a reason that I will need to investigate before this PR can be merged. 

@gziolo As a compromise would you be okay with removing the part of this test that downloads external themes? We can still test switching between themes that exist in the environment (i.e. any `twenty*` theme). Backup option is just removing this test -- any issues with the utils would surface in test environments anyway.

Reply:
I don't think I ever considered external themes here, it should be perfectly fine to use one of many bundled themes.

Reply:
👍  I will update this week.